re-write

xDimGG opened the pull request:
tell me if you dont like anything ( or dont like everything )

lloti committed:
re-write

tell me if you dont like anything

lloti committed:
check for album

lloti committed:
switched pause and unpause

lloti committed:
hecking gus

cernodile commented on app.js:
```diff
@@ -1,140 +1,65 @@
-  });
-}
+try {
+	const path = process.platform === 'win32' ? 'C:\\Windows\\System32\\drivers\\etc\\hosts' : '/hosts/etc';
```
Not everyone has it on C drive, plus there's one more win32 system that allows to change install path + it's `/etc/hosts` for Linux

Phineas commented:
Looks good, could you add the timer pause feature when the song gets paused tho? Also, when I do anything on Spotify, it takes around 10 seconds to update on Discord.

cernodile commented on app.js:
```diff
@@ -1,140 +1,65 @@
+	activity.smallImageKey = image;
+	if (song) {
+		activity.details = `🎵  ${song.title}`;
+		activity.state = `👤  ${song.artist.name}`;
```
What if `song.artist` is undefined, never happens on normal songs, but would happen on local files.

cernodile commented on app.js:
```diff
@@ -1,140 +1,65 @@
+		activity.details = `🎵  ${song.title}`;
+		activity.state = `👤  ${song.artist.name}`;
+		activity.largeImageText = `🔗  ${song.id}`;
+		activity.smallImageText = `💿  ${song.album.id}`;
```
Same as `song.artist`, this can cause troubles on local files.

cernodile commented on app.js:
```diff
@@ -1,140 +1,65 @@
-  log(`Song state updated (playing: ${song.playing})`)
+client.on('ready', () => {
+	log(`Connected to Discord! (${cfg.id})`);
+	setInterval(() => client.setActivity(activity), 15e3);
```
15e3?

xDimGG replied:
15e3 = 15000 = 15 * 10 ^ 3

cernodile replied:
What if the user changes song? It would create quite a noticeable gap.

cernodile commented on spotify.js:
```diff
@@ -1,303 +1,82 @@
+
+class Song {
+	constructor(data) {
+		this.title = data.track.track_resource.name;
```
Add checks for data incase of local files.

cernodile reviewed (changes requested):
Please fix `/hosts/etc`, make checks for win32 platform to discover installation path, create additional data checks in case of local files.

lloti committed:
Buncha Fixes

Previous method of setting the presence every 15 seconds was removed, instead it just checks how long it's been since you last updated it.

lloti committed:
nicer looking + more fallbacks

cernodile commented on app.js:
```diff
@@ -11,7 +11,7 @@ const spotify = new Spotify();
  **/
 try {
-	const path = process.platform === 'win32' ? 'C:\\Windows\\System32\\drivers\\etc\\hosts' : '/hosts/etc';
+	const path = process.platform === 'win32' ? 'C:\\Windows\\System32\\drivers\\etc\\hosts' : '/etc/hosts';
```
What about win32, not every installation is at C:\ nor \Windows\. Use %SystemRoot% to your advantage.

cernodile commented on app.js:
```diff
@@ -25,16 +25,37 @@ const activity = {
+			time = Date.now();
+		}, timeLeft);
+	} else {
+		client.setActivity(activity);
```
"else it'll set a timeout until we can", yet it sets activity either way?

cernodile replied:
nvm, OP clarified this to me in Discord. I misunderstood.

cernodile reviewed (changes requested):
You've mitigated the primary issues. There are some still problems remaining, good job though.

lloti committed:
Flipped code & used %SYSTEMDIRECTORY%

lloti committed:
*SystemRoot

cernodile reviewed (approved):
Other than spotify.js, this rewrite seems clean enough now.

Phineas commented:
Will test & merge tomorrow :)

Phineas commented:
Timer still doesn't disappear when paused for me & theres still a delay when I do anything.
